sale_order_type/models/sale.py

```diff
@@ -73,6 +74,16 @@ def _prepare_invoice(self):
             res["sale_type_id"] = self.type_id.id
         return res
 
+    @api.constrains('company_id', 'type_id.company_id')
+    def _constrains_company(self):
+        if self.company_id not in self.type_id.company_id:
+            raise exceptions.ValidationError("Error : Company and Type No Matching !!!")
+
+    @api.onchange('company_id', 'type_id.company_id')
+    def check_company(self):
```

Comment:
```suggestion
    def _check_company(self):
```

Reply:
I would add to change the method name to `_check_company_order_type` to be more understandable.